build/validate.sh

```diff
@@ -1,14 +1,17 @@
 #!/bin/bash
-echo -e "\nChecking go fmt bosun.org/..."
-GOFMTOUT=$(go fmt -s bosun.org/...);
-if [ "$GOFMTOUT" != '' ]; then 
-    echo "The following files need gofmt -s:";
-    echo "$GOFMTOUT";
-    exit 1;
+echo -e "\nChecking gofmt -l -s -w for all folders that don't start with . or _"
+cd $GOPATH/src/bosun.org
+GOFMTRESULT=0
+GOFMTOUT=$(find . -maxdepth 1 -type d -iregex './[^._].*' -exec gofmt -l -s -w {} +);
+if [ "$GOFMTOUT" != '' ]; then
+    echo "The following files need 'gofmt -l -s -w':"
```

Comment:
we need to use -l to get the list of files that changed:

> C:\Code\Go\src\bosun.org [travis_fmt +0 ~1 -0]> gofmt -?
> flag provided but not defined: -?
> usage: gofmt [flags] [path ...]
>   -cpuprofile="": write cpu profile to this file
>   -d=false: display diffs instead of rewriting files
>   -e=false: report all errors (not just the first 10 on different lines)
>   -l=false: list files whose formatting differs from gofmt's
>   -r="": rewrite rule (e.g., 'a[b:len(a)] -> a[b:]')
>   -s=false: simplify code
>   -w=false: write result to (source) file instead of stdout